src/elements/flag/flag.js

```diff
@@ -0,0 +1,19 @@
+(function () {
+    'use strict';
+
+    angular
+        .module('semantic.ui.elements.flag', [])
+        .directive('smFlag', smFlag);
+
+    function smFlag($compile) {
+        return {
+            restrict: 'E',
+            replace: true,
+            scope: { value: '='},
+            link: function (scope, element) {
+                element.html('').append($compile('<i class="flag ' + scope.value + '"></i>')(scope));
```

Comment:
Another thing, the `$compile()` is probably not really needed here --- The template looks pretty static to me.